dotnet/samples/GettingStartedWithProcesses/Utilities/MermaidRenderer.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Reflection;
+using PuppeteerSharp;
+
+namespace Utilities;
+
+/// <summary>
+/// Renders Mermaid diagrams to images using Puppeteer-Sharp.
+/// </summary>
+public static class MermaidRenderer
+{
+    /// <summary>
+    /// Generates a Mermaid diagram image from the provided Mermaid code.
+    /// </summary>
+    /// <param name="mermaidCode"></param>
+    /// <param name="filename"></param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    public static async Task GenerateMermaidImageAsync(string mermaidCode, string filename)
+    {
+        // Locate the current assembly's directory
+        string? assemblyPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
```

Comment:
>string? assemblyPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location); [](http://example.com/codeflow?start=8&length=87)

since this is a tool that could be used anywhere perhaps this can be more flexible. 

Example:

1. user may provide absolute filepath -> this should be used -> make sure dir path exists - throw exception if it doesnt
2. user provides only filename: `someimage.png` then it seems ok to use the Assembly path

Reply:
good suggestion :)